adal/src/main/java/com/microsoft/aad/adal/TokenCacheItem.java

```diff
@@ -106,6 +111,9 @@ private TokenCacheItem(final String authority, final AuthenticationResult authen
         mRawIdToken = authenticationResult.getIdToken();
         mRefreshtoken = authenticationResult.getRefreshToken();
         mFamilyClientId = authenticationResult.getFamilyClientId();
+        if (authenticationResult.getExtendedExpiresOn() != null) {
```

Comment:
why checking for null instead of just doing assign? And why not checking for null for all the other variables? This is confusing.